### PR TITLE
feat(soft-assertions): Implement soft assertions for playwright-java

### LINKED ISSUE #819: [Feature] Bring SoftAssertions to Java
Node version has soft assertions now. Can we get soft assertions for Java too?

### PR DESCRIPTION
**Pull request by uchagani**
Resolves #819

### CODE REVIEW

File: playwright/src/main/java/com/microsoft/playwright/impl/APIResponseAssertionsImpl.java
Comment (uchagani) [inline playwright/src/main/java/com/microsoft/playwright/impl/APIResponseAssertionsImpl.java:40]: I change the return value to the Impl for `not` as [@yury-s suggested ](https://github.com/microsoft/playwright-java/pull/1340#discussion_r1291649659)

File: playwright/pom.xml
Comment (uchagani) [inline playwright/pom.xml:78]: I hope adding Mockito as a test dependency is okay.  For soft assertion test, all we need to verify is that we are calling the correct underlying assertion method instead.  So I'm using Mockito for that.

File: playwright/src/main/java/com/microsoft/playwright/BrowserContext.java
Comment (uchagani) [inline playwright/src/main/java/com/microsoft/playwright/BrowserContext.java:134]: the project won't build until these methods are implemented.  I stubbed them out for my purposes but then removed the stubs before creating this PR.

Comment (yury-s) [reply playwright/src/main/java/com/microsoft/playwright/BrowserContext.java:134]: I'm rolling the driver in https://github.com/microsoft/playwright-java/pull/1362 with implementation of all the new APIs but SoftAssertions. You'll need to revert my change in ApiGenerator.java to make it produce SoftAssertions.java

**Review by yury-s — APPROVED** [review]

File: playwright/src/test/java/com/microsoft/playwright/TestSoftAssertions.java
Comment (uchagani) [reply playwright/src/test/java/com/microsoft/playwright/TestSoftAssertions.java:56]: good point, i'll just remove the check

File: playwright/src/test/java/com/microsoft/playwright/TestSoftLocatorAssertions.java
Comment (uchagani) [reply playwright/src/test/java/com/microsoft/playwright/TestSoftLocatorAssertions.java:87]: i wrote these tests because i had fat fingered a couple of the calls.  i think they do provide value when new assertions are added and leaves less up to chance.  I moved away from a lot of direct tests of the assertions as I do in `TestSoftAssertions` because i found myself duplicating the same error messages as the actual assertion tests.  If those messages ever change, we would need to update in more places.

File: pom.xml
Comment (uchagani) [reply pom.xml:52]: java 8 requires v4 of mockito https://github.com/mockito/mockito/releases/tag/v5.0.0

**Comment by uchagani** [comment]
@yury-s this is ready for another review please.

**Review by yury-s — APPROVED** [review]

**Comment by yury-s** [comment]
@uchagani looks good, once the test bots finish we can merge it. Thanks for implementing this!

**Comment by yury-s** [comment]
Docker tests failed because we now need to copy mockito jars similar to [these lines](https://github.com/microsoft/playwright-java/blob/fa75e29fcf73fa752d87bc5cba6d11d5c2591ecd/tools/test-local-installation/create_project_and_run_tests.sh#L12-L14). Alternatively, you can skip those tests when running in docker.

**Comment by uchagani** [comment]
@yury-s excluded mockito tests from docker, can i get another approval please

File: tools/test-local-installation/pom.xml
Comment (uchagani) [inline tools/test-local-installation/pom.xml:1]: @yury-s since it was a compilation error and not a test error, i had to add mockito as a test dependency in this pom.xml file.  it seems we have a different pom.xml for docker test runs.  Once i added mockito here as a test dependency everything works.

**Review by yury-s — APPROVED** [review]

**Comment by uchagani** [comment]
@yury-s do we need to re-run these failures?  seems like they only failed on mac and windows but don't look related to my changes

**Comment by yury-s** [comment]
No need to rerun, all look unrelated to your PR, I'll merge it.

**Pull request merged**
